src/main.rs

```diff
@@ -20,50 +23,66 @@ struct Cli {
+    )]
+    token: String,
+    #[arg(long, help = "Path to repository root. Defaults to current directory.")]
+    repo_root: Option<String>,
+    #[arg(long, help = "Value to override URL of repository.")]
+    repo_url: Option<String>,
+    #[arg(long, help = "Value to override SHA of repository head.")]
+    repo_head_sha: Option<String>,
+    #[arg(long, help = "Value to override branch of repository head.")]
+    repo_head_branch: Option<String>,
+    #[arg(long, help = "Value to override commit epoch of repository head.")]
+    repo_head_commit_epoch: Option<String>,
+    #[arg(
+        long,
+        value_delimiter = ',',
+        help = "Comma separated list of custom tag=value pairs."
+    )]
+    tags: Vec<String>,
+    #[arg(long, help = "Print files which will be uploaded to stdout.")]
+    print_files: bool,
+    #[arg(long, help = "Run metrics CLI without uploading to API.")]
+    dry_run: bool,
+}
+
+#[derive(Args, Clone, Debug)]
+struct TestArgs {
+    #[command(flatten)]
+    upload_args: UploadArgs,
+    #[arg(
+        required = false,
+        allow_hyphen_values = true,
+        trailing_var_arg = true,
+        help = "Test command to invoke."
+    )]
+    command: Vec<String>,
+}
+
 #[derive(Debug, Subcommand)]
 enum Commands {
-    #[clap(name = "upload")]
-    Upload {
-        #[arg(
-            long,
-            required = true,
-            value_delimiter = ',',
-            help = "Comma-separated list of glob paths to junit files."
-        )]
-        junit_paths: Vec<String>,
-        #[arg(long, help = "Organization url slug.")]
-        org_url_slug: String,
-        #[arg(
-            long,
-            required = true,
-            env = "TRUNK_API_TOKEN",
-            help = "Organization token. Defaults to TRUNK_API_TOKEN env var."
-        )]
-        token: String,
-        #[arg(long, help = "Path to repository root. Defaults to current directory.")]
-        repo_root: Option<String>,
-        #[arg(long, help = "Value to override URL of repository.")]
-        repo_url: Option<String>,
-        #[arg(long, help = "Value to override SHA of repository head.")]
-        repo_head_sha: Option<String>,
-        #[arg(long, help = "Value to override branch of repository head.")]
-        repo_head_branch: Option<String>,
-        #[arg(long, help = "Value to override commit epoch of repository head.")]
-        repo_head_commit_epoch: Option<String>,
-        #[arg(
-            long,
-            value_delimiter = ',',
-            help = "Comma separated list of custom tag=value pairs."
-        )]
-        tags: Vec<String>,
-        #[arg(long, help = "Print files which will be uploaded to stdout.")]
-        print_files: bool,
-        #[arg(long, help = "Run metrics CLI without uploading to API.")]
-        dry_run: bool,
-    },
+    Upload(UploadArgs),
+    Test(TestArgs),
 }
 
-const DEFAULT_API_ADDRESS: &str = "https://api.trunk.io/v1/metrics/createBundleUpload";
```

Comment:
I think this should be safe to rollout. I don't think any of our customers are setting this externally.